specifications/secure-launch-specification.rst

```diff
@@ -0,0 +1,719 @@
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_entry_policy {
+        struct slr_entry_hdr hdr;
+        u16 revision;
+        u16 nr_entries;
+        /* policy_entries[] */
+    };
+  
+
+DRTM Policy Entry
+"""""""""""""""""
+
+A policy entry represents an entity that the SL Entry is being requested to
+measure. As an SL Entry is able to measure an attribute of the launch
+environment, that attribute will be published as an entity type. A generic
+"unspecified" entity type is also available for measuring a range of memory.
+
+:pcr: PCR to store the measurement.
+:entity_type: Identifies the entity type of the entry.
+:flags: Flag field to store state for this entry.
+:entity: The address to measure.
+:size: The size of entity if not flagged as implicit.
+:evt_info: Label to be recorded in TPM Event Log.
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_policy_entry {
+        u16 pcr;
+        u16 entity_type;
+        u16 flags;
+        u16 reserved;
+        u64 entity;
+        u64 size;
+        char evt_info[TPM_EVENT_INFO_LENGTH];
+    };
+
+DRTM Policy Entry Entity Types
+''''''''''''''''''''''''''''''
+
+The list of valid entity types for DRTM Policy entries.
+
+.. code-block:: c
+    :linenos: 1
+
+    #define SLR_ET_UNSPECIFIED  0x0000
+    #define SLR_ET_SLRT         0x0001
+    #define SLR_ET_BOOT_PARAMS  0x0002
+    #define SLR_ET_SETUP_DATA   0x0003
+    #define SLR_ET_CMDLINE      0x0004
+    #define SLR_ET_EFI_MEMMAP   0x0005
+    #define SLR_ET_RAMDISK      0x0006
+    #define SLR_ET_TXT_OS2MLE   0x0010
+    #define SLR_ET_UNUSED       0xffff
+
+DRTM Policy Entry Flags
+'''''''''''''''''''''''
+
+The list of valid flags for DRTM Policy entries.
+
+.. code-block:: c
+    :linenos: 1
+
+    #define SLR_POLICY_FLAG_MEASURED	0x1
+    #define SLR_POLICY_IMPLICIT_SIZE	0x2
+
+Intel TXT Platforms
+~~~~~~~~~~~~~~~~~~~
+
+When on Intel platforms specific information needs to be conveyed to Secure
+Launch.
+
+Intel TXT Info
+^^^^^^^^^^^^^^
+
+:REQUIRED: This entry **MUST** be present on Intel platforms.
+
+Intel TXT requires for the pre-launch environment to pass MSR and MTRR state
+across to the post-launch environment.
+
+:tag: SLR_ENTRY_INTEL_INFO
+:saved_misc_enable_msr: Saved MSR values
+:saved_bsp_mtrrs: Saved BSP MTRRs
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_entry_intel_info {
+        struct slr_entry_hdr hdr;
+        u64 saved_misc_enable_msr;
+        struct txt_mtrr_state saved_bsp_mtrrs;
+    };
+
+Saved MTRR State
+""""""""""""""""
+
+:code:`struct txt_mtrr_state`
+
+:default_mem_type: The default memory type for regions not covered by an MTRR
+:mtrr_vcnt: Number of variable MTRR pairs in the mtrr_vcnt array
+:mtrr_pair: Array of variable MTRR pairs to restore post launch
+
+:code:`struct txt_mtrr_pair`
+
+:mtrr_physbase: Physical base address for variable MTRR
+:mtrr_physmask: Physical mask for the variable MTRR
+
+.. code-block:: c
+    :linenos: 1
+
+    struct txt_mtrr_pair {
```

Comment:
I missed this while working on the doc with Daniel but we should prefix these with slr_ to be consistent